### PR TITLE
chore(deps): updating Spectral packages so that Arazzo and AsyncAPI

### LINKED ISSUE #216: Update to use latest spectral core
Use latest spectral core with vscode

### LINKED ISSUE #234: Support AsyncAPI 2.6.0 in vscode extension
**spectral CLI supports AsyncAPI 2.6, vscode extension does not**
Would like the vscode extension to also support 2.6 as the spectral CLI supports this.

### PR DESCRIPTION
**Pull request by frankkilcommins**
Fixes #234
Fixes #216 

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

Updating Spectral packages so that Arazzo and AsyncAPI linting is included in the VSCode extension.
Also bump node versions on github actions.

### CODE REVIEW

**Review by YOU54F — APPROVED** [review]
Might want to bump the minimum version of node that is supported in the node engines field, if any of the new packages mandate versions greater

https://github.com/stoplightio/vscode-spectral/blob/a5c5dcc112d3e2d4142ce359aef3cf729c9ba575/package.json#L113

Apart from that looks good, will be good to allow users of this extension the benefit of validating AsyncAPI and the new Arazzo based documents

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -10,12 +10,12 @@ jobs:
...
       matrix:
         os: [macos-latest, ubuntu-latest, windows-latest]
-        node-version: [^12.13]
+        node-version: [18.x]
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
 
-      - uses: actions/setup-node@v3.5.1
+      - uses: actions/setup-node@v4.0.1
```
Comment (YOU54F) [inline .github/workflows/test.yml:18]: v4 will track minor/patch releases

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -10,12 +10,12 @@ jobs:
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest, windows-latest]
-        node-version: [^12.13]
+        node-version: [18.x]
```
Comment (YOU54F) [inline .github/workflows/test.yml:13]: we may want to add a matrix to test across node versions, or at least cover the minimum specified in the node engines field.

If it can't run on an older version, and there is no suitable workaround, you can them bump the node engines version along with (ideally) a major version bump is the standard practise

Comment (frankkilcommins) [reply .github/workflows/test.yml:13]: yep - one we can add in soon

File: .github/workflows/test.yml (hunk 1 above)
Comment (frankkilcommins) [reply .github/workflows/test.yml]: bumped to v4

**Review by mnaumanali94 — APPROVED** [review]
Generally looks good. Might want to add a test for arazzo or could just create an issue for it as there are none for asyncapi either.

**Comment by mnaumanali94** [comment]
Also should update the readme to add Arazzo and new Async version

**Comment by frankkilcommins** [comment]
> Generally looks good. Might want to add a test for arazzo or could just create an issue for it as there are none for asyncapi either.

@mnaumanali94 - I created #241  for this request

**Comment by frankkilcommins** [comment]
> Also should update the readme to add Arazzo and new Async version

@mnaumanali94 - nice spot. I've updated the ReadMe.
